### ISSUE TITLE
Let `copy` support a function

### ISSUE THREAD

**Issue by SamVerschueren** [valor-software/ng2-dragula#608]
Currently, copying an object is done with `JSON.parse(JSON.stringify(object))`. This works for JSON objects but not if you want to work with classes or if you want custom copy functionality.

So currently, the only thing that you can set is a boolean like this.

```js
dragulaService.setOptions('bag', {
      copy: true
});
```

This should still be possible, but as an extra it would be nice if you could do this

```js
dragulaService.setOptions('bag', {
      copy: object => new MyCustomObject(object)   //=> copy constructor
});
```

I can do a PR if you want. Just let me know what you think.

**Comment by cormacrelf** [comment]
I'm for it. Probably call it `copyModel` to fit the naming scheme, but I'll accept that PR.

**Comment by ccordenier** [comment]
Hi

I would be very interested in this feature, but I didn't find any related PR. I can contribute too, but I need some help to find a clean way of doing it.

To me it's 'only' a matter of replacing 

`let dropElmModel = notCopy ? sourceModel[dragIndex] :JSON.parse(JSON.stringify(sourceModel[dragIndex]));`

To use the copyModel option. 

But I don't see a way to get access to drake options without passing the original options object as a parameter.

**Comment by cormacrelf** [comment]
#490 was first.

**Comment by cormacrelf** [comment]
Closed in 78f3f9f1917dc765ff9cca6ee5eb25d4775be3de
